braze/client.py

```diff
@@ -81,41 +92,80 @@ def user_delete(self, external_ids, appboy_ids):
         if appboy_ids:
             payload['appboy_ids'] = appboy_ids
 
-        return self.__create_request(payload=payload, request_type=self.REQUEST_POST)
+        return self.__create_request(payload=payload)
 
-    def __create_request(self, payload, request_type):
-        self.headers = {
-            'Content-Type': 'application/json',
-        }
+    def __create_request(self, payload):
 
         payload['api_key'] = self.api_key
 
-        response = {}
-
+        response = {'errors': []}
         try:
-            if request_type == self.REQUEST_POST:
-                r = self.requests.post(self.request_url, data=json.dumps(payload), headers=self.headers)
-                response = r.json()
-                response['status_code'] = r.status_code
-                if response['message'] == 'success' and 'errors' not in response:
+            r = self._post_request_with_retries(payload)
+            response.update(r.json())
+            response['status_code'] = r.status_code
+
+            message = response['message']
+            if message == 'success' or message == 'queued':
+                if not response['errors']:
                     response['success'] = True
+                else:
+                    # Non-Fatal errors
+                    pass
 
-        except RequestException as e:
-            # handle all requests HTTP exceptions
-            response = {'client_error': str(e)}
-        except Exception as e:
-            # handle all exceptions which can be on API side
-            response = {'client_error': (str(e) + '. Response: ' + r.text)}
+        except (
+                RequestException,
+                BrazeRateLimitError,
+                BrazeInternalServerError,
+        ) as e:
+            response['errors'].append(str(e))
```

Comment:
This is preexisting from the fork but I don't think this a great pattern. Better if we just raise the exception and let the caller handle it.

Reply:
I considered both approaches. No strong opinions on either. We could let the caller handle the exception as well which would reduce the amount of code here but it would increase the amount of code the consumer of the client has to write every time it calls an api method (the consumer will always have to catch exceptions and handle them). We can think more about this ahead as we start using this library in our codebase and change if throwing exceptions seems more useful than trickling up errors.

Reply:
Talked offline, to be addressed in follow up PR.